config/migrate.ts

```diff
@@ -0,0 +1,191 @@
+import { i18n } from '../utils/lang';
+import { logger } from '../lib/logger';
+
+const i18nKey = 'config.migrate';
+
+export function getConfig(
+  useHiddenConfig?: boolean,
+  configPath?: string
+): Partial<CLIConfig> | null {
+  if (useHiddenConfig) {
+    return CLIConfiguration.config;
+  }
+  return config_DEPRECATED.loadConfig(configPath);
+}
+
+export function getConfigPath(
+  configPath?: string,
+  useHiddenConfig = false
+): string | null {
+  if (useHiddenConfig) {
+    return getConfigFilePath();
+  }
+  return config_DEPRECATED.getConfigPath(configPath);
+}
+
+export function configFileExists(
+  useHiddenConfig?: boolean,
+  configPath?: string
+): boolean {
+  return useHiddenConfig
+    ? newConfigFileExists()
+    : Boolean(config_DEPRECATED.getConfigPath(configPath));
+}
+
+export function deleteConfigFile(useHiddenConfig = false): void {
+  if (useHiddenConfig) {
+    newDeleteConfigFile();
+  }
+  config_DEPRECATED.deleteConfigFile();
+}
+
+function writeGlobalConfigFile(updatedConfig: CLIConfig_NEW): void {
+  const updatedConfigJson = JSON.stringify(updatedConfig);
+  createEmptyConfigFile({}, true);
+  loadConfig('');
+
+  try {
+    writeConfig({ source: updatedConfigJson });
+    deleteConfigFile();
+  } catch (error) {
+    deleteEmptyConfigFile();
+
+    throw new Error(
+      i18n(`${i18nKey}.errors.writeConfig`, { configPath: GLOBAL_CONFIG_PATH }),
+      { cause: error }
+    );
+  }
+}
+
+export function migrateConfig(
+  deprecatedConfig: CLIConfig_DEPRECATED | null
+): void {
+  if (!deprecatedConfig) {
+    throw new Error(i18n(`${i18nKey}.errors.noDeprecatedConfig`));
+  }
+  const { defaultPortal, portals, ...rest } = deprecatedConfig;
+  const updatedConfig = {
+    ...rest,
+    defaultAccount: defaultPortal,
+    accounts: portals
+      .filter(({ portalId }) => portalId !== undefined)
+      .map(({ portalId, ...rest }) => ({
+        ...rest,
+        accountId: portalId!,
+      })),
+  };
+  writeGlobalConfigFile(updatedConfig);
+}
+
+function mergeConfigPropertes(
+  globalConfig: CLIConfig_NEW,
+  deprecatedConfig: CLIConfig_DEPRECATED
+): CLIConfig_NEW {
+  const propertiesToCheck: Array<keyof Partial<CLIConfig>> = [
+    'defaultCmsPublishMode',
+    'httpTimeout',
+    'allowUsageTracking',
+    'env',
+  ];
+  const conflicts: Array<{
+    property: keyof CLIConfig_NEW;
+    oldValue: boolean | string | number | CmsPublishMode | Environment;
+    newValue: boolean | string | number | CmsPublishMode | Environment;
+  }> = [];
+
+  propertiesToCheck.forEach(prop => {
+    if (prop in globalConfig && prop in deprecatedConfig) {
+      if (globalConfig[prop] !== deprecatedConfig[prop]) {
+        conflicts.push({
+          property: prop,
+          oldValue: deprecatedConfig[prop]!,
+          newValue: globalConfig[prop]!,
+        });
+      }
+    } else {
+      // @ts-expect-error TODO
+      globalConfig[prop] = deprecatedConfig[prop];
```

Comment:
Is this resolved?

Reply:
I've added a note, but unfortunately, I don't believe we can resolve this without doing something gross. I've added a note to say that we can't reconcile the `CLIConfig_NEW` and `CLIConfig_DEPRECATED` types. If you have any ideas, I'm all ears 👂